EOS-8749: Revert global indexes on create bucket error

dsurnin opened the pull request:
Signed-off-by: Dmitrii Surnin <dmitrii.surnin@seagate.com>

dsurnin commented:
http://eos-jenkins.colo.seagate.com/job/S3server/job/s3-github-test/89/

johnbent commented:
@dsurnin , are you getting the reviews you need on this?

bkirunge7 commented on server/s3_bucket_metadata_v1.cc:
```diff
@@ -482,12 +469,17 @@ void S3BucketMetadataV1::save_replica() {
+    }
+    this->handler_on_failed();
   } else {
-    state = S3BucketMetadataState::failed;
```
will this state value will be used at other places

dsurnin replied:
yes, this is an operation status

evgeniy-brazhnikov reviewed (approved):
Good for me

bkirunge7 reviewed (commented):
@dsurnin see if you can add a System test for this use KV tool check if index is exists

dsurnin commented:
> 
> 
> @dsurnin see if you can add a System test for this use KV tool check if index is exists

I already modified existing tests and added a couple of new

bkirunge7 commented:
> > @dsurnin see if you can add a System test for this use KV tool check if index is exists
> 
> I already modified existing tests and added a couple of new

I see unit tests added, no system tests, like create a bucket using aws cli with fault injection enabled, and see if it does cleanup in index table.  If its feasible then fine.

dsurnin commented:
> 
> 
> > > @dsurnin see if you can add a System test for this use KV tool check if index is exists
> > 
> > 
> > I already modified existing tests and added a couple of new
> 
> I see unit tests added, no system tests, like create a bucket using aws cli with fault injection enabled, and see if it does cleanup in index table. If its feasible then fine.

I didn't bring any new functionality, just reused old one, already proved and tested. In unit tests I checked that it is properly called. I think it should be enough.
Approach you suggest will require several server restarts - we do not use FI bucket creation now in tests, so stop prev, reconfigure startup, run single test for this particular scenario, stop and reconfigure as it was previous,start and continue testing. It will increase build time and test logic since we do not assume any restarting now.
